app/Models/Characters/Character.php

```diff
@@ -87,4 +87,34 @@ private function abilityScoreToModifier($ability_score)
     {
         return (int) floor(($ability_score - 10) / 2);
     }
+
+    public function getArmorClassAttribute()
+    {
+        $armor_class = 10;
+
+        $armor_skill_modifiers = $this->armor->modifiers()
+            ->where('type', 'stat')
+            ->where('code', 'armor_class')
+            ->get();
+
+        foreach ($armor_skill_modifiers as $modifier) {
+            if ($modifier->bonus) {
+                if (is_numeric($modifier->bonus)) {
+                    $armor_class += (int) $modifier->bonus ?? 0;
+                } else {
+                    $armor_class += $this->{$modifier->bonus} ?? 0;
+                }
+            }
+
+            $armor_class = max(
+                $modifier->min ?? 0,
+                min(
+                    $modifier->max ?? INF,
+                    $armor_class
+                )
+            );
+        }
```

Comment:
Better todo something like:
```php
return $base_ac + $this->armor
  ->modifiers()
  ->where('type', 'stat')
  ->where('code', 'armor_class')
  ->get()
  ->reduce(function ($carry, $modifier) {
    // The logic from inside the loop using $carry instead of $armour_class
  })
```

Reply:
> I don't feel the need for it. The modifiers only change character stats, and they are applied during computation of those stats.

Mmm, I know what you mean but it'll make the code more extensible and clear as you can hide the 
```php
if ($modifier->bonus) {
                if (is_numeric($modifier->bonus)) {
                    $armor_class += (int) $modifier->bonus ?? 0;
                } else {
                    $armor_class += $this->{$modifier->bonus} ?? 0;
                }
            }
```
in a getter like
```php
    function getBonusAttribute($value) {
       if (is_int($value)) {
           return $value;
       }

        return data_get(
            $this,
            $value,
            0,
        );
    }
```

Reply:
Unfortunately I'd forgotten about the ManyToMany from Character -> Items -> Modifiers so the getter wouldn't work but I just changed it to a method instead where you pass data. In this case data is the character.

```php
$modifier->get_bonus(
  $character,
);
```

This also means we can support the dot notation in the future if needed as you can do:
```php
$modifier->get_bonus(
  [
    'character' => $character,
    'something' => $somthing,
  ]
);
```
where the db bonus column instead looks like:
```character.dexterity```
```something.anotherValue```